spark/v3.2/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/RewriteUpdateTable.scala

```diff
@@ -56,24 +58,41 @@ object RewriteUpdateTable extends RewriteRowLevelCommand {
     case u @ UpdateIcebergTable(aliasedTable, assignments, cond, None) if u.resolved && u.aligned =>
       EliminateSubqueryAliases(aliasedTable) match {
         case r @ DataSourceV2Relation(tbl: SupportsRowLevelOperations, _, _, _, _) =>
-          val operation = buildRowLevelOperation(tbl, UPDATE)
-          val table = RowLevelOperationTable(tbl, operation)
-          val updateCond = cond.getOrElse(Literal.TrueLiteral)
-          val rewritePlan = operation match {
-            case _: SupportsDelta =>
-              buildWriteDeltaPlan(r, table, assignments, updateCond)
-            case _ if SubqueryExpression.hasSubquery(updateCond) =>
-              buildReplaceDataWithUnionPlan(r, table, assignments, updateCond)
-            case _ =>
-              buildReplaceDataPlan(r, table, assignments, updateCond)
+          rewriteUpdateTable(u, r, tbl)
+        case v: View =>
+          val relations = v.children.collect { case r: DataSourceV2Relation if r.table.isInstanceOf[SparkTable] =>
```

Comment:
I think there's mAh e  types of table within Spark than just `SparkTable` now but I'm not sure if that matters (cached tables used for some procédures I thubj).